django/conf/__init__.py

```diff
@@ -185,6 +194,19 @@ def __init__(self, settings_module):
                 setattr(self, setting, setting_value)
                 self._explicit_settings.add(setting)
 
+        # Normalize SECRET_KEY/SECRET_KEYS.
+        # * When one value is provided, map to the other.
+        # * Neither is unusual but allowed if not accessing the value, in a
+        #   script, for example. Both values are validated on first-access.
+        if self.SECRET_KEY and not self.SECRET_KEYS:
+            self.SECRET_KEYS = [self.SECRET_KEY]
+        elif self.SECRET_KEYS and not self.SECRET_KEY:
+            self.SECRET_KEY = self.SECRET_KEYS[0]
+
```

Comment:
In case it's relevant, this patches up normal usages of settings - that is, using a settings module and `Settings` - but leaves `UserSettingsHolder` without the same shoring up (although I may be mis-reading the bigger picture, apologies if so). 
That is, if I do `django.conf.settings.configure(SECRET_KEY='...')` I won't end up with a `SECRET_KEYS` where `[0] == SECRET_KEY`, I think? Less of a problem going forward (users will just add the key), only an issue for extant usages getting upgraded.
It'll still error with `ImproperlyConfigured` correctly on accessing the lazy setting, mind. 
I know some things are intentionally left to the caller (`tzset` shenanigans are omitted). Is this one of those?

Reply:
Hey @kezabelle. Thanks for the comments. Thinking about both now. 

This one: 

> ...leaves UserSettingsHolder without the same shoring up...

I'm thinking that a mapping in `__getattr__` might be the way to go — _We missed the one lookup, let's try the other_ — I think we probably need that for compatibility around the ecosystem. ... 🤔 — I'll add a test case and see.